librz/asm/arch/hexagon/hexagon_arch.c

```diff
@@ -646,24 +752,100 @@ static HexInsnContainer *hex_to_new_pkt(HexState *state, const HexInsnContainer
  * \return HexInsnContainer* Pointer to the copied instruction container on the heap.
  */
 static HexInsnContainer *hex_add_to_stale_pkt(HexState *state, const HexInsnContainer *new_hic) {
-	HexPkt *p = hex_get_stale_pkt(state);
-	hex_clear_pkt(p);
+	HexPkt *pkt = hex_get_stale_pkt(state);
+	hex_clear_pkt(pkt);
 
 	HexInsnContainer *hic = hexagon_alloc_instr_container();
 	hex_copy_insn_container(hic, new_hic);
-	rz_list_insert(p->bin, 0, hic);
+	rz_list_insert(pkt->bin, 0, hic);
 
-	p->last_instr_present |= is_last_instr(hic->parse_bits);
-	p->pkt_addr = new_hic->addr;
+	pkt->last_instr_present |= is_last_instr(hic->parse_bits);
+	pkt->pkt_addr = new_hic->addr;
 	// p->is_valid = true; // Setting it true also detects a lot of data as valid assembly.
-	p->last_access = rz_time_now();
-	hex_set_pkt_info(&state->rz_asm, hic, p, 0, false);
-	if (p->last_instr_present) {
-		make_next_packet_valid(state, p);
+	pkt->last_access = rz_time_now();
+	hex_set_pkt_info(&state->rz_asm, hic, pkt, 0, false);
+	if (pkt->last_instr_present) {
+		make_next_packet_valid(state, pkt);
 	}
 	return hic;
 }
 
+#if RZ_BUILD_DEBUG
+static char desc_letter_hic(const HexInsnContainer *hic) {
+	char desc = ' ';
+	if (!hic) {
+		desc = ' ';
+	} else if (hic->is_duplex) {
+		desc = hic->bin.sub[0]->identifier != HEX_INS_INVALID_DECODE ? 'v' : 'i';
+		desc = hic->pkt_info.last_insn ? 'l' : desc;
+	} else {
+		desc = hic->bin.insn->identifier != HEX_INS_INVALID_DECODE ? 'v' : 'i';
+		desc = hic->pkt_info.last_insn ? 'l' : desc;
+	}
+	return desc;
+}
+#endif
+
+static void print_state_pkt(const HexState *state, st32 index, HexBufferAction action, const HexInsnContainer *new_hic) {
+#if RZ_BUILD_DEBUG
+	ut32 oldest = 7;
+	ut32 newest = 0;
+	ut64 min_time = 0xffffffffffffffff;
+	ut64 max_time = 0;
+	for (int i = 0; i < HEXAGON_STATE_PKTS; ++i) {
+		const HexPkt *pkt = &state->pkts[i];
+		if (pkt->last_access < min_time) {
+			min_time = pkt->last_access;
+			oldest = i;
+		}
+		if (pkt->last_access > max_time) {
+			max_time = pkt->last_access;
+			newest = i;
+		}
+	}
+	RZ_LOG_DEBUG("╭─────┬──────────────┬─────┬──────────────────┬───────────────╮\n");
```

Comment:
We should have general API for that, inside RzTable.

Reply:
Couldn't do the last column which is split into 4 and makes it easier to read IMHO. Do you think it is still better? Then I change it. Though it is only debug printing.

Reply:
It was just a remark, not a requirement. But it would be nice to figure out what's missing in RzTable.